test/integration/scheduler/queue_test.go

```diff
@@ -277,6 +277,26 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			},
 			wantRequeuedPods: sets.New("pod1"),
 		},
```

Comment:
Can you improve `Pod rejected by the NodeAffinity plugin is requeued when a Node is updated` to see if match -> match update is ignored correctly, which makes this PR's benefit visible?